### PR TITLE
Ensured that observers are added only once

### LINKED ISSUE #927: [Bug][iOS][MediaElement] NSInternalInconsistencyException
### Description
A `MediaElement` can't be deallocated on older iOS versions (<13). It leads to an `NSInternalInconsistencyException`. I don't know why it is working on newer iOS versions, maybe they handle this kind of errors more gracefully now. The problem is, that Observers are added more often than they are disposed.

### Steps to Reproduce

1. Add a `MediaElement` to any Page
2. Deploy to an iOS Device with iOS <13
2. Ensure the Page gets removed from the NavigationStack (eg. resetting the `App.MainPage`)

### Expected Behavior
App should continue to run

### Actual Behavior
App freezes or crashes. 
```
Foundation.MonoTouchException: Objective-C exception thrown.  Name: NSInternalInconsistencyException Reason: An instance 0x61000041e660 of class AVPlayerItem was deallocated while key value observers were still registered with it. Current observation info: <NSKeyValueObservationInfo 0x61000342dde0> (
<NSKeyValueObservance 0x610000245280: Observer: 0x610002637240, Key path: status, Options: <New: YES, Old: NO, Prior: NO> Context: 0x610002637240, Property: 0x610000245400>
)
```

### Basic Information

- Version with issue:  1.0.2
- Last known good version: -
- Platform Target Frameworks: 
  - iOS:  14
- Affected Devices: All iOS Devices with iOS <13. Tested with iOS 10-14 (always last minor) 

### Reproduction Link
https://github.com/Huaba93/XamarinIssues/tree/master/MediaElementCrashIOS

### PR DESCRIPTION
**Pull request by Huaba93**
### Bugs Fixed ###

- Fixes #927 

### API Changes ###
None

<!-- List all API changes here (or just put None), example:

Added: 
 
- `string Class.Property { get; set; }`
- `void Class.Method();`

Changed:

 - `object Cell.OldPropertyName` => `object Cell.NewPropertyName`
 
If there is an entirely new API, then you can use a more verbose style:

```csharp
public static class NewClass {
    public static int SomeProperty { get; set; }
    public static void SomeMethod(string value);
}
``` -->

### PR Checklist ###
<!-- Please check all the things you did here and double-check that you got it all, or state why you didn't do something -->

- [ ] Has tests (if omitted, state reason in description)
- [ ] Has samples (if omitted, state reason in description)
- [x] Rebased on top of main at time of PR
- [x] Changes adhere to coding standard
<!-- If at all possible, please update/add the documentation on the repo below. We would very much appreciate that. If you are unable to, please consider at least opening an issue on the repo below so we know that Docs still need to be adjusted/created. Thanks! <3 -->
- [ ] Updated [documentation](https://github.com/MicrosoftDocs/xamarin-communitytoolkit)

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/xamarin/XamarinCommunityToolkit?pullRequest=928) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/xamarin/XamarinCommunityToolkit?pullRequest=928) before we can accept your contribution.<br/><br/>:x: Huaba93 [sign now](https://cla.dotnetfoundation.org/xamarin/XamarinCommunityToolkit?pullRequest=928)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/xamarin/XamarinCommunityToolkit?pullRequest=928) it.</sub>

**Comment by jfversluis** [comment]
@peterfoot you have any opinions on this?

**Comment by jfversluis** [comment]
@Huaba93 I think I saw that the CLA was signed earlier, right? We require this to be able to take contributions from you. Could you double-check everything is in order? Thanks!

**Comment by peterfoot** [comment]
> @peterfoot you have any opinions on this?

This looks good, improves clarity and deals with the issue where source is changed multiple times causing multiple observers.

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/MediaElement/iOS/MediaElementRenderer.ios.cs [hunk 1]
```diff
@@ -26,9 +26,9 @@ public class MediaElementRenderer : ViewRenderer<ToolKitMediaElement, UIView>
 		protected IDisposable rateObserver;
 		protected IDisposable volumeObserver;
 		bool idleTimerDisabled = false;
+		AVPlayerItem item;
```
Comment (pictos) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Views/MediaElement/iOS/MediaElementRenderer.ios.cs:29]: Can you rename this to a better that describes the type? Could be `playerItem` or something like that

**Review by pictos — COMMENTED** [review]
That's looks good, just a small suggestion

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/MediaElement/iOS/MediaElementRenderer.ios.cs (hunk 1 above)
Comment (Huaba93) [reply src/CommunityToolkit/Xamarin.CommunityToolkit/Views/MediaElement/iOS/MediaElementRenderer.ios.cs:29]: I wanted to make as few changes as possible, but you are right, this name would be much better. 

I will rename it.

**Comment by Huaba93** [comment]
@jfversluis CLA is signed again. I do not know why this was reset. 

@pictos Suggestion was applied.

**Review by pictos — APPROVED** [review]

**Pull request merged**